### PR TITLE
[PATRO23-154] add design time dbcontex creation

### PR DESCRIPTION
**Pull request by DawidBski**
https://tracker.intive.com/jira/browse/PATRO23-154

### CODE REVIEW

File: src/api/app/ExampleDbContextFactory.cs
Comment (pzajaczkowski) [inline src/api/app/ExampleDbContextFactory.cs:1]: In docs i found sentence: "If a class implementing this interface is found in either the same project as the derived DbContext or in the application's startup project, the tools bypass the other ways of creating the DbContext and use the design-time factory instead.". So i think we should move dbcontext factory to its module infrastructure project (same project as ExampleDbContext) to keep things organized. Correct me if i'm wrong.

File: src/api/app/appsettings.Development.json [hunk 1]
```diff
@@ -7,6 +7,6 @@
 	},
 	"ConnectionStrings": {
 		"DockerDb": "Server=db; Database=IntivePatronage-2023-DB; User=SA; Password=S3cur3P@ssW0rd!; TrustServerCertificate=true",
-		"AppDb": "Server=(localdb)\\MSSQLLocalDB; Database=IntivePatronage-2023-DB; Trusted_Connection=true; TrustServerCertificate=true"
+		"MigrationsDb": "Server=localhost,1433; Database=IntivePatronage-2023-DB; User=SA; Password=S3cur3P@ssW0rd!; TrustServerCertificate=true"
 	}
-}
+}
```
Comment (pzajaczkowski) [inline src/api/app/appsettings.Development.json:12]: no new line at the end of file

File: src/api/app/appsettings.json [hunk 2]
```diff
@@ -9,6 +9,6 @@
 	"AllowedHosts": "*",
 	"ConnectionStrings": {
 		"DockerDb": "Server=db; Database=IntivePatronage-2023-DB; User=SA; Password=S3cur3P@ssW0rd!; TrustServerCertificate=true",
-		"AppDb": "Server=(localdb)\\MSSQLLocalDB; Database=IntivePatronage-2023-DB; Trusted_Connection=true; TrustServerCertificate=true"
+		"MigrationsDb": "Server=localhost,1433; Database=IntivePatronage-2023-DB; User=SA; Password=S3cur3P@ssW0rd!; TrustServerCertificate=true"
 	}
-}
+}
```
Comment (pzajaczkowski) [inline src/api/app/appsettings.json:14]: same thing

**Review by pzajaczkowski — CHANGES_REQUESTED** [review]

File: src/api/app/ExampleDbContextFactory.cs
Comment (DawidBski) [reply src/api/app/ExampleDbContextFactory.cs:1]: You're right i just forgot about that. It's cool that you read docs. Nice that you noticed it!

File: src/api/app/appsettings.json (hunk 2 above)
Comment (DawidBski) [reply src/api/app/appsettings.json:14]: end_line

File: src/api/app/appsettings.Development.json (hunk 1 above)
Comment (DawidBski) [reply src/api/app/appsettings.Development.json:12]: end_line_light

File: src/api/app/ExampleDbContextFactory.cs
Comment (DawidBski) [reply src/api/app/ExampleDbContextFactory.cs:1]: Done :)

File: src/api/app/appsettings.json (hunk 2 above)
Comment (pzajaczkowski) [reply src/api/app/appsettings.json:14]: ok XD haven't seen that before.

**Review by pzajaczkowski — APPROVED** [review]

File: src/api/app/appsettings.json (hunk 2 above)
Comment (DawidBski) [reply src/api/app/appsettings.json:14]: I was looking for that about 10-15 min :)

File: src/api/app/appsettings.Development.json [hunk 3]
```diff
@@ -7,6 +7,6 @@
 	},
 	"ConnectionStrings": {
 		"DockerDb": "Server=db; Database=IntivePatronage-2023-DB; User=SA; Password=S3cur3P@ssW0rd!; TrustServerCertificate=true",
-		"AppDb": "Server=(localdb)\\MSSQLLocalDB; Database=IntivePatronage-2023-DB; Trusted_Connection=true; TrustServerCertificate=true"
+		"MigrationsDb": "Server=localhost,1433; Database=IntivePatronage-2023-DB; User=SA; Password=S3cur3P@ssW0rd!; TrustServerCertificate=true"
```
Comment (mateuszgiedzinski) [inline src/api/app/appsettings.Development.json:10]: Is it good idea to have password here?

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

File: src/api/app/appsettings.Development.json (hunk 3 above)
Comment (DawidBski) [reply src/api/app/appsettings.Development.json:10]: For production purpose isn't but for development I believe it's fine.

Comment (DawidBski) [reply src/api/app/appsettings.Development.json:10]: For a production purpose I would use Environment Variables.

File: src/api/app/appsettings.json [hunk 4]
```diff
@@ -9,6 +9,6 @@
 	"AllowedHosts": "*",
 	"ConnectionStrings": {
 		"DockerDb": "Server=db; Database=IntivePatronage-2023-DB; User=SA; Password=S3cur3P@ssW0rd!; TrustServerCertificate=true",
-		"AppDb": "Server=(localdb)\\MSSQLLocalDB; Database=IntivePatronage-2023-DB; Trusted_Connection=true; TrustServerCertificate=true"
+		"MigrationsDb": "Server=localhost,1433; Database=IntivePatronage-2023-DB; User=SA; Password=S3cur3P@ssW0rd!; TrustServerCertificate=true"
```
Comment (mateuszgiedzinski) [inline src/api/app/appsettings.json:12]: Still secret for db context is present. .

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

**Review by mateuszgiedzinski — APPROVED** [review]

**Comment by mateuszgiedzinski** [comment]
When you will merge commit please sellect option to Merge and squash commits.  There are to many commits for such small change. Don't forget to add to your commits [Task number] - commit description. 
I saw that you forgot about that.

**Pull request merged**